### PR TITLE
CloudFlare DNS driver

### PR DESCRIPTION
**Pull request by Kami**
New driver for CloudFlare DNS (WIP).

TODO:
- [x] Support for record management (create, edit, delete)
- [x] Tests
- [ ] Pagination support

### CODE REVIEW

**Comment by Kami** [comment]
Alright, added the tests and fixed some things.

I will go ahead and merge it and implement pagination in a future PR (driver is already functional without pagination as long as you have < page size number of zones and records).

File: libcloud/test/dns/fixtures/cloudflare/nul.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
```
Comment (tonybaloney) [inline libcloud/test/dns/fixtures/cloudflare/nul.json:1]: @Kami nul is a reserved file name in Windows, so anyone trying to test or use this on Windows is going to have a lot a bad time.. Please can you rename it

Comment (Kami) [inline libcloud/test/dns/fixtures/cloudflare/nul.json:1]: Even if it has an extension (aka `nul.json`)?

Comment (Kami) [inline libcloud/test/dns/fixtures/cloudflare/nul.json:1]: It looks like it, that's unfortunate :P

Good catch - will rename it.

Comment (Kami) [inline libcloud/test/dns/fixtures/cloudflare/nul.json:1]: Would make it easier if we could also automatically run tests on Windows, but Travis sadly doesn't support that (yet).

**Comment by dol** [comment]
@Kami Thank you for the driver. Any reason the driver uses a old version of the Cloudflare API -> https://www.cloudflare.com/docs/client-api.html .
CloudFlare is removing support for this API on November 9th, 2016.

**Comment by Kami** [comment]
@dol I think that when I added it, new version of the API wasn't there yet (or I missed it).

**Comment by dol** [comment]
Ok. Sorry, the question was unfortunate.
Any plans switching over to the new API version?

**Comment by Kami** [comment]
I don't have any plans, but contributions are welcome (you can also ping / ask CloudFlare people).

**Comment by dol** [comment]
I'm not that frequent in Python. Might be better to call your friends.

**Pull request merged**